### PR TITLE
handle error depth

### PR DESCRIPTION
**Pull request by lightsing**

### CODE REVIEW

File: bus-mapping/src/evm/opcodes.rs [hunk 1]
```diff
@@ -285,7 +288,11 @@ fn fn_gen_error_state_associated_ops(error: &ExecError) -> Option<FnGenAssociate
         ExecError::InsufficientBalance => Some(CallOpcode::<7>::gen_associated_ops),
         ExecError::WriteProtection => Some(ErrorWriteProtection::gen_associated_ops),
         ExecError::ReturnDataOutOfBounds => Some(ErrorReturnDataOutOfBound::gen_associated_ops),
-
+        ExecError::Depth => {
+            let op = geth_step.op;
+            assert!(op.is_call());
```
Comment (lispc) [inline bus-mapping/src/evm/opcodes.rs:293]: if !op.is_call() {
    unimplemented!("ErrDepth in create will be implemented later");
}

File: bus-mapping/src/evm/opcodes/callop.rs [hunk 2]
```diff
@@ -170,7 +171,7 @@ impl<const N_ARGS: usize> Opcode for CallOpcode<N_ARGS> {
             .unwrap_or(false);
         // TODO: What about transfer for CALLCODE?
         // Transfer value only for CALL opcode, insufficient_balance = false.
-        if call.kind == CallKind::Call && !insufficient_balance {
+        if call.kind == CallKind::Call && !insufficient_balance && is_depth_ok {
```
Comment (lispc) [inline bus-mapping/src/evm/opcodes/callop.rs:174]: set "!insufficient_balance && is_depth_ok" as "is_precheck_ok" similar to scroll/develop

File: zkevm-circuits/src/evm_circuit/execution/callop.rs [hunk 3]
```diff
@@ -155,9 +155,14 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
         );
         let is_insufficient_balance =
             LtWordGadget::construct(cb, &caller_balance_word, &call_gadget.value);
+        // depth < 1025
+        let is_depth_ok = LtGadget::construct(cb, depth.expr(), 1025.expr());
+        let is_error_depth = not::expr(is_depth_ok.expr());
+        let is_insufficient_balance_or_error_depth =
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs:161]: rename like scroll/develop

File: zkevm-circuits/src/evm_circuit/execution/callop.rs [hunk 4]
```diff
@@ -989,4 +1008,65 @@ mod test {
...
+        .unwrap();
+
+        CircuitTestBuilder::new_from_test_ctx(ctx)
+            .params(CircuitsParams {
+                max_rws: 300000,
+                ..Default::default()
+            })
+            .run();
+
+        // assert_eq!(
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs:1049]: what is the comments? needed?

File: zkevm-circuits/src/evm_circuit/execution/callop.rs [hunk 5]
```diff
@@ -446,6 +455,9 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
             step.rw_indices[5],
         ]
         .map(|idx| block.rws[idx].call_context_value());
+        let error_depth = depth.low_u64() > 1024;
+        self.is_depth_ok
+            .assign(region, offset, F::from(depth.low_u64()), F::from(1025))?;
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs:460]: change error_depth to is_error_depth to be more consistent with other names like is_insufficient?

File: zkevm-circuits/src/evm_circuit/execution/callop.rs [hunk 6]
```diff
@@ -290,7 +294,11 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
         });
 
         cb.condition(
-            not::expr(no_callee_code) * not::expr(is_insufficient_balance.expr()),
+            and::expr(&[
+                not::expr(no_callee_code),
+                not::expr(is_insufficient_balance.expr()),
+                is_depth_ok.expr(),
+            ]),
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs:301]: and::expr(&[not::expr(no_callee_code), is_precheck_ok.expr())) ?

**Review by lispc — APPROVED** [review]

**Review by ChihChengLiang — APPROVED** [review]
LGTM

**Pull request merged**